### PR TITLE
Change `on_subgroups` action to use the GAP interface

### PR DESCRIPTION
**Pull request by StevellM**
To work with (orbits and) stabilizers of subgroups in `TorQuadModule`, I implemented a very naive approach which was quite expensive, using action on lattices. Since the automorphisms for `TorQuadModule` come from GAP, we change the definition of the action to use the GAP action (I do not have the details here, but on relevant examples we are faster by a factor 8). I also needed `is_conjugate_with_data` for the action on subgroups.

As a remark: a submodule of a `TorQuadModule` do not "remember its parent" (sad). So to work with actions on submodules (on the GAP side), we need to remember the embedding of the underlying subgroup. This is why the `stabilizer` and `is_conjugate_with_data` methods take homomorphism(s) as input(s).

As for the general group interface, I added the action `on_subgroups`: I am ok with suggestions to modify it/change the name, as long as it allows me to make my stabilizer computations faster.

I also deleted what I thought would be useful for incoming new features (i.e. `compute_stab`): it is not used at the end so better removing it. Note that what I have removed was not exported.

### CODE REVIEW

File: experimental/QuadFormAndIsom/src/embeddings.jl [hunk 1]
```diff
@@ -267,17 +246,17 @@ function _subgroups_orbit_representatives_and_stabilizers(Vinq::TorQuadModuleMor
     filter!(s -> is_invariant(fV, s[2]), subs)
   end
 
-  subs = TorQuadModule[s[1] for s in subs]
-  m = gset(O, _on_subgroups, subs)
+  to_gap = get_attribute(O, :to_gap)
+  to_oscar = get_attribute(O, :to_oscar)
+
+  qgap = codomain(to_gap)
+  sgap = typeof(qgap)[sub(qgap, elem_type(qgap)[to_gap(q(lift(s[2](a)))) for a in gens(s[1])])[1] for s in subs]
```
Comment (simonbrandhorst) [inline experimental/QuadFormAndIsom/src/embeddings.jl:253]: Better create the subgroups directly in gap and only translate representatives to Oscar. 
But in any case the number of subgroups is usually to large to put them in memory.

**Review by simonbrandhorst — APPROVED** [review]

File: experimental/QuadFormAndIsom/src/embeddings.jl (hunk 1 above)
Comment (StevellM) [reply experimental/QuadFormAndIsom/src/embeddings.jl:253]: I am not sure how to keep track on the form on the GAP side but yes, there are still some possible improvements to be made here. I guess that in practice, this part of the code will be called on small enough examples where we should not have too many (stable) subgroups.

Comment (simonbrandhorst) [reply experimental/QuadFormAndIsom/src/embeddings.jl:253]: Do you need the form in the orbit/stabliser computation?

Comment (simonbrandhorst) [reply experimental/QuadFormAndIsom/src/embeddings.jl:253]:
```

    subs = collect(submodules(V; order = ord))
    filter!(s -> is_invariant(fV, s[2]), subs)
```
does not see the form

Comment (StevellM) [reply experimental/QuadFormAndIsom/src/embeddings.jl:253]: Ah no right indeed, I forgot that the submodules are created as subgroups which inherit the form from the bigger module. So indeed yes, we could do everything on the GAP side. I can try it now and see to make a change in this PR. Thanks!

Comment (StevellM) [reply experimental/QuadFormAndIsom/src/embeddings.jl:253]: I actually I do not know if it is better. On the GAP side, from what I see, we cannot enumerate subgroups of a given order, or stable subgroups. Which is what we use here in Hecke. I am afraid that on the GAP side we will have to many subgroups: ok we can filter them anyway but if our group has hundreds of subgroups but only 1 of the order we want, Hecke will do things better here (see https://github.com/oscar-system/Oscar.jl/issues/2835)

Comment (simonbrandhorst) [reply experimental/QuadFormAndIsom/src/embeddings.jl:253]: o.k.

**Pull request merged**